RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGeneratorFromL1TkMu.cc

```diff
@@ -0,0 +1,413 @@
+/*  \class L2MuonSeedGeneratorFromL1TkMu
+ *
+ *   L2 muon seed generator:
+ *   Transform the L1TkMuon informations in seeds
+ *   for the L2 muon reconstruction
+ *   (mimicking L2MuonSeedGeneratorFromL1T)
+ *
+ *    Author: H. Kwon
+ *    Modified by M. Oh
+ */
+
+// Class Header
+#include "RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGeneratorFromL1TkMu.h"
+
+// Framework
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h"
+#include "TrackingTools/TrajectoryParametrization/interface/CurvilinearTrajectoryError.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "TrackingTools/TrajectoryState/interface/FreeTrajectoryState.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+#include "TrackingTools/KalmanUpdators/interface/Chi2MeasurementEstimator.h"
+#include "TrackingTools/DetLayers/interface/DetLayer.h"
+
+#include "RecoMuon/TrackingTools/interface/MuonServiceProxy.h"
+#include "RecoMuon/TrackingTools/interface/MuonPatternRecoDumper.h"
+
+#include "DataFormats/Math/interface/LorentzVector.h"
+
+using namespace std;
+using namespace edm;
+using namespace l1t;
+
+// constructors
+L2MuonSeedGeneratorFromL1TkMu::L2MuonSeedGeneratorFromL1TkMu(const edm::ParameterSet &iConfig)
+    : theSource(iConfig.getParameter<InputTag>("InputObjects")),
+      thePropagatorName(iConfig.getParameter<string>("Propagator")),
+      theL1MinPt(iConfig.getParameter<double>("L1MinPt")),
+      theL1MaxEta(iConfig.getParameter<double>("L1MaxEta")),
+      theMinPtBarrel(iConfig.getParameter<double>("SetMinPtBarrelTo")),
+      theMinPtEndcap(iConfig.getParameter<double>("SetMinPtEndcapTo")),
+      theMinPL1Tk(iConfig.getParameter<double>("MinPL1Tk")),
+      theMinPtL1TkBarrel(iConfig.getParameter<double>("MinPtL1TkBarrel")),
+      useOfflineSeed(iConfig.getUntrackedParameter<bool>("UseOfflineSeed", false)),
+      useUnassociatedL1(iConfig.getParameter<bool>("UseUnassociatedL1")),
+      matchingDR(iConfig.getParameter<std::vector<double>>("MatchDR")),
+      etaBins(iConfig.getParameter<std::vector<double>>("EtaMatchingBins")) {
+  muCollToken_ = consumes<l1t::TkMuonCollection>(theSource);
```

Comment:
this sholud also be initialised in the member initialisation, above:
```suggestion
  muCollToken_(consumes<l1t::TkMuonCollection>(theSource))
```